Unbreaking by ploxiln

ploxiln opened the pull request:
a continuation of #1, you can see my additional commits at the end

Commit:
Factored out tornado bits and refactored persistence/loading logic to be more robust.

Commit:
Version bump.

Commit:
Cleaned up CountingBloomFilter and added the beginnings of a new test suite.

Commit:
Added better comments.

Commit:
Added some integration tests for counting blooms.

Commit:
Started to add unit tests for the timing blooms.

Commit:
Fleshed out the test suite for timing bloom filters.

Commit:
Made data persistence optional.

Commit:
Made ticker default to NoOpTicker if not provided.

Commit:
Updated travis config to use py.test instead of the tornado testing tools.

Commit:
Trying again with the travis config.

Commit:
Trying yet again with the travis config.

Commit:
Trying another travis config.

Commit:
Improved .gitignore file.

Commit:
Added tests and fixes for ScalingTimingBloomFilter.

Commit:
Added integration tests for ScalingTimingBloomFilter.

Commit:
Removed unused files.

Commit:
Added ticker tests.

Commit:
Version bump.

Commit:
Updated persistence to be a bit more robust.

Commit:
Version bump.

Commit:
Addressed some review comments.

ploxiln committed:
fix flush_data() path check, get rid of can_persist()

ploxiln committed:
bring back fuggetaboutit.benchmark

notice that arguments to TimingBloomFilter work differently now

ploxiln committed:
don't build with profiling enabled

it doesn't work unless it's also enabled for linking, which it's not

I don't think it's off by default, and I don't think we want profiling
output to be created every run by default.

ploxiln committed:
don't need to link in libc

it's libc, it's automatic

ploxiln commented:
I'll admit I threw those compiler and linker flags changes in there without much awareness of python conventions or practices.

ploxiln committed:
better if TimingBloomFilter takes capacity first, like the rest

mynameisfiber commented:
Hey @ploxiln... so I have a couple of issues with this pull request.  The main two are:
- There are no more tests to make sure that the error rate is what we expect nor any to make sure we are expiring the correct values
- A lot of the new api requires persistence or requires the data location to be set at initialization.  I would rather this be provided on the save operation (ie: have `.save()` take in a data path argument and never have it be a property of the bloom object).  A result of this would be that `meta_filename` and `blooms_path` would be created dynamically on save and also wouldn't need to be object properties.

Other than that, I think a lot of the changes here are really awesome.

ploxiln committed:
counting_bloom_filter: refactor save() and flush_data()

now they all take data_path arguments

also delete a bunch of unit tests that really just tested implementation details

ploxiln commented:
I made the data_path a parameter to `save()` and `flush_data()` on CountingBloomFilter and TimingBloomFilter. It can also be set at initialization time. I'll do ScalingTimingBloomFilter next.

I deleted a bunch of unit tests that mocked internal functions and system calls and expected a very specific implementation... I really hate that style of testing. I'll add a few more integration tests that include decaying, saving, loading, and checking accuracy, as future step 2.

ploxiln committed:
ScalingTimingBloomFilter: clean up docstring

ploxiln committed:
optimizations: simplify some bit ops

temp is only compared to 0, so no need to shift it

what we're really doing is bitwise-or, not add

ploxiln committed:
remove imports that are unused after unit tests were trimmed

ploxiln committed:
remove unused import of struct in experiment

ploxiln committed:
ScalingTimingBloomFilter: save takes optional data_path argument

ploxiln committed:
beginnings of full scaling timing bloom integration test
